packages/app/src/firebase.js

```diff
@@ -53,25 +54,26 @@ export function FirebaseAuthProvider({ children }) {
             displayName: user.displayName,
             email: user.email,
             photoURL: user.photoURL,
-            username,
             createdAt: firebase.firestore.FieldValue.serverTimestamp(),
           }
-          setUser(signedInUser)
-          try {
-            const ref = await firebase.firestore()
-              .collection('users')
-              .doc(user.uid)
-              .get()
 
-            if (!ref.exists) {
-              await firebase.firestore()
-                .collection('users')
-                .doc(signedInUser.uid)
-                .set(signedInUser)
-            }
-          } catch (e) {
-            console.log('Some error with firebase', e)
+          if (username) {
+            signedInUser[username] = username
+          }
+
+          const ref = await firebase.firestore()
+            .collection('users')
+            .doc(user.uid)
+            .get()
+
+          if (!ref.exists) {
+            console.log(signedInUser)
+            await firebase.firestore()
+              .collection('users')
+              .doc(signedInUser.uid)
+              .set(signedInUser)
           }
+          setUser(signedInUser)
           setIsSignedIn(true)
```

Comment:
I think you could just do:
```js
          if (username) {
            signedInUser[username] = username
          }

          await firebase.firestore()
            .collection('users')
            .doc(user.uid)
            .set(signedInUser, { merge: true })

          setUser(signedInUser)
          setIsSignedIn(true)
```

When the user signs in for the first time it will be created.
When the use come back you will update it's data in the database (if he changes his photoURL, displayName or even his username...)

But I think that it's not necessary to store these data, you can always access it through `firebase` module:
```
// user contains: uid, displayName, displayName, photoURL
const user = firebase.auth().currentUser
```

Reply:
This comment will be fixed in #35